elephant/test/test_phase_analysis.py

```diff
@@ -370,5 +373,266 @@ def testPhaseLockingValue_raise_Error_if_trial_lengths_are_different(self):
             self.simple_y, self.simple_z)
 
 
+class WeightedPhaseLagIndexTestCase(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        np.random.seed(73)
+        cls._generate_datasets_for_ground_truth_testing()
```

Comment:
I am not sure about this flow. The two `.mat` files saved by this function are used in the other scripts, to generate the CSVs to check the results against the other packages. But every time this test is run, new `.mat` files are generated, so the precomputed ground truth will not correspond. How is everything integrated into the testing pipeline? 

If that is the case, one alternative would be to have this function returning arrays, using it outside the test to generate the `.mat` for the scripts using FieldTrip and MNE (i.e., before running the scripts for MNE or Fieldtrip, call a script that computes the source `.mat` files used).
 
Then, when instantiating this test, call the function again and check if the arrays are the same as the ones stored in the `.mat` files that were used for the ground truth CSVs.